Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -184,25 +184,62 @@ namespace
         }
     }
 
+    void SetXRGamepadObjectData(Napi::Object& jsInputSource, Napi::Object& jsGamepadObject, xr::System::Session::Frame::InputSource& inputSource)    
+    {
+        auto env = jsInputSource.Env();
+        //Set Gamepad Object
+
+        auto gamepadButtons = Napi::Array::New(env, inputSource.GamepadObject.Buttons.size());
+        for (size_t i = 0; i < inputSource.GamepadObject.Buttons.size(); i++)
+        {
+            auto gamepadButton = Napi::Object::New(env);
+            auto napiGamepadPressed = Napi::Boolean::New(env, inputSource.GamepadObject.Buttons[i].Pressed);
+            auto napiGamepadTouched = Napi::Boolean::New(env, inputSource.GamepadObject.Buttons[i].Touched);
+            auto napiGamepadValue = Napi::Number::New(env, inputSource.GamepadObject.Buttons[i].Value);
+            gamepadButton.Set("pressed", napiGamepadPressed);
+            gamepadButton.Set("touched", napiGamepadTouched);
+            gamepadButton.Set("value", napiGamepadValue);
+            gamepadButtons.Set(static_cast<int>(i), gamepadButton);
+        }
+        jsGamepadObject.Set("buttons", gamepadButtons);
+
+        auto gamepadAxes = Napi::Array::New(env, inputSource.GamepadObject.Axes.size());
+        for (size_t i = 0; i < inputSource.GamepadObject.Axes.size(); i++)
+        {
+            auto napiGamepadAxesValue = Napi::Number::New(env, inputSource.GamepadObject.Axes[i]);
+            gamepadAxes.Set(static_cast<int>(i), napiGamepadAxesValue);
+        }
+        jsGamepadObject.Set("axes", gamepadAxes);
+        jsInputSource.Set("gamepad", jsGamepadObject);
+    }
+
     Napi::ObjectReference CreateXRInputSource(xr::System::Session::Frame::InputSource& inputSource, Napi::Env& env)
     {
         constexpr std::array<const char*, 2> HANDEDNESS_STRINGS{
             "left",
             "right"};
         constexpr const char* TARGET_RAY_MODE{"tracked-pointer"};
-
         auto jsInputSource = Napi::Object::New(env);
```

Comment:
nit: was this whitespace shift intentional?

Reply:
nope it is not, i can reverse that

Reply:
nit: there's still new whitespace